src/hotspot/share/prims/unsafe.cpp

```diff
@@ -242,6 +242,11 @@ class MemoryAccess : StackObj {
     return normalize_for_read(*addr());
   }
 
+  // we use this method at some places for writing to 0 e.g. to cause a crash;
+  // ubsan does not know that this is the desired behavior
+#if defined(__clang__) || defined(__GNUC__)
+__attribute__((no_sanitize("undefined")))
+#endif
```

Comment:
Can we hide this in a macro like `SUPPRESS_UBSAN_WARNING`? If it turns out we need to do this in a few places then it will look nicer.

Reply:
Sounds like a good idea.  See also the discussion about `ATTRIBUTE_NO_UBSAN`  here  https://github.com/openjdk/jdk/pull/19597

Reply:
I like the idea of an encapsulating macro as well.

Reply:
I guess  ATTRIBUTE_NO_UBSAN will do if we have a precedent for that naming.